pkg/operator/controller/extension/reconciler.go

```diff
@@ -109,14 +113,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, fmt.Errorf("error retrieving virtual cluster client set: %w", err)
 	}
 
-	if extension.DeletionTimestamp != nil {
-		return reconcile.Result{}, r.delete(ctx, log, virtualClusterClientSet.Client(), extension)
+	if extension.DeletionTimestamp != nil || extension.Annotations[AnnotationKeyDisableAdmission] == "true" {
+		if extension.Annotations[AnnotationKeyDisableAdmission] == "true" {
+			log.Info("Disabling admission controller because of annotation " + AnnotationKeyDisableAdmission)
+		}
+		return reconcile.Result{}, r.delete(ctx, log, virtualClusterClientSet.Client(), extension, extension.DeletionTimestamp == nil)
```

Comment:
That's a good example how this annotation makes the code more complex. I really vote for removing this annotation handling an let users rathe work with the `spec` in the API directly.